corehq/apps/cleanup/management/commands/swap_duplicate_xforms.py

```diff
@@ -0,0 +1,158 @@
+            dest='dry_run',
+            default=False,
+            help="Don't do the actual modifications, but still log what would be affected."
+        ),
+        make_option(
+            '--no-input',
+            action='store_true',
+            dest='no_input',
+            default=False,
+            help='Skip important confirmation warnings.'
+        ),
+    )
+
+    def handle(self, *args, **options):
+
+        self.dups_by_domain = {}
+        # e.g.
+        #   {
+        #       "foo-domain": {
+        #           "an-original-id": {"a-duplicate-id", "another-duplicate-id"}
+        #       }
+        #   }
+
+        dry_run = options.get("dry_run", True)
+        if not options.get('no_input', False) and not dry_run:
+            confirm = raw_input(
+                u"""
+                Are you sure you want to swap the given XFormInstances for duplicates?
+                This is NOT a dry run. y/N?
+                """
+            )
+            if confirm != "y":
+                print "\n\t\tSwap duplicates cancelled."
+                return
+
+        ids_file_path = args[0].strip()
+        log_path = args[1].strip()
+
+        num_with_1_dup = 0
+        num_with_multi_dup = 0
+        num_with_no_dup = 0
+
+        with open(log_path, "w") as log_file:
+            with open(ids_file_path, "r") as lines:
+
+                total_lines = 0
+                for _ in lines:
+                    total_lines += 1
+                lines.seek(0)
+
+                for i, (domain, bad_xform_id) in enumerate(l.split() for l in lines):
+                    self._print_progress(i, total_lines)
+                    duplicates = self.get_duplicates(domain, bad_xform_id)
+                    if len(duplicates) == 1:
+                        num_with_1_dup += 1
+                        self.swap_doc_types(log_file, bad_xform_id, list(duplicates)[0], dry_run)
+                    elif len(duplicates) > 1:
+                        num_with_multi_dup += 1
+                        self.log_too_many_dups(log_file, bad_xform_id, duplicates)
+                    else:
+                        num_with_no_dup += 1
+
+        print "Found {} forms with no duplicates".format(num_with_no_dup)
+        print "Found {} forms with one duplicate".format(num_with_1_dup)
+        print "Found {} forms with multiple duplicates".format(num_with_multi_dup)
+
+    def get_duplicates(self, domain, xform_id):
+
+        if domain not in self.dups_by_domain:
+            self.populate_dup_map(domain)
+        return self.dups_by_domain[domain].get(UUID(xform_id), {})
+
+    def populate_dup_map(self, domain):
+        self.dups_by_domain[domain] = defaultdict(set)
+        dups = XFormInstance.view(
+            'couchforms/all_submissions_by_domain',
+            startkey=[domain, 'XFormDuplicate', '2016'],
+            endkey=[domain, "XFormDuplicate", {}],
+            reduce=False,
+            include_docs=True
+        )
```

Comment:
neither `orig_id` or `problem` is indexed in ES, so couldn't search there. Iterating over so many dups seems a bit inefficient, but the whole script still only takes a minute or so (without saving), so seems ok to me.